nixos/modules/services/monitoring/karma.nix

```
@@ -0,0 +1,179 @@
+{ config, pkgs, lib, ... }:
+with lib;
+let
+  cfg = config.services.karma;
+
+in
+{
+  options.services.karma = {
+    enable = mkEnableOption (mdDoc "the Karma dashboard service.");
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.karma;
+      defaultText = literalExpression "pkgs.karma";
+      description = mdDoc ''
+        The Karma package that should be used.
+      '';
+    };
+
+    configFile = mkOption {
+      type = with types; nullOr path;
+      default = null;
+      description = mdDoc ''
+        Provides Karma with a raw config file instead of a nix-generated file.
+      '';
+      example = "/etc/karma/karma.conf";
+    };
+
+    environment = mkOption {
+      type = types.attrs;
+      default = {};
+      description = mdDoc ''
+        Additional environment variables to provide to karma.
+      '';
+      example = {
+        ALERTMANAGER_URI = "https://alertmanager.example.com";
+        ALERTMANAGER_NAME= "single";
+      };
+    };
+
+    openFirewall = mkOption {
+      type = types.bool;
+      default = false;
+      description = mdDoc ''
+        Whether to open ports in the firewall needed for karma to function.
+      '';
+    };
+
+    extraOptions = mkOption {
+      type = with types; listOf str;
+      default = [];
+      description = mdDoc ''
+        Extra command line options.
+      '';
+      example = [
+        "--alertmanager.timeout 10s"
+      ];
+    };
+
+    address = mkOption {
+      type = types.str;
+      default = "0.0.0.0";
+      description = mdDoc ''
+        Hostname or IP to listen on.
+      '';
+      example = "127.0.0.1";
```

```suggestion
      example = "0.0.0.0";
```
Does it support ``[::]``?

---

They've not specified it here https://github.com/prymitive/karma/blob/main/docs/CONFIGURATION.md#listen

But they are using the golang http.Server so it probably would support that. The upstream default is 0.0.0.0 so should I keep it that way?